samza-core/src/main/scala/org/apache/samza/system/StreamMetadataCache.scala

```diff
@@ -80,7 +80,7 @@ class StreamMetadataCache (
           val systemAdmin = systemAdmins
             .getOrElse(systemName, throw new SamzaException("Cannot get metadata for unknown system: %s" format systemName))
           val streamToMetadata = if (partitionsMetadataOnly && systemAdmin.isInstanceOf[ExtendedSystemAdmin]) {
-            systemAdmin.asInstanceOf[ExtendedSystemAdmin].getSystemStreamPartitionCounts(systemStreams.map(_.getStream).asJava, cacheTTLms)
+            systemAdmin.asInstanceOf[ExtendedSystemAdmin].getSystemStreamMetadata(systemStreams.map(_.getStream).asJava)
```

Comment:
This makes the `if` part of the conditional identical to the `else`. 

I think the original intent was for the `if` to use the faster "getSystemStreamPartitionCounts" method. So this will introduce a performance regression for any code paths that invoke this method with partitionsMetadataOnly==true.  We can't do that. 

I'm in favor of removing the partitionsMetadataOnly argument altogether and changing the callers that used that flag to use the "getSystemStreamPartitionCounts" method directly.